modules/lang-mustache/src/internalClusterTest/java/org/elasticsearch/script/mustache/SearchTemplateIT.java

```diff
@@ -189,6 +191,69 @@ public void testIndexedTemplateClient() throws Exception {
         assertNull(getResponse.getSource());
     }
 
+    public void testBadTemplate() {
+
+        // This template will produce badly formed json if given a multi-valued `text_fields` parameter,
+        // as it does not add commas between the entries. We test that it produces a 400 json parsing
+        // error both when used directly and when used in a render template request.
+
+        String script = """
+            {
+              "script": {
+                "lang": "mustache",
+                "source": ""\"
+                {
+                  "query": {
+                    "multi_match": {
+                      "query": "{{query_string}}",
+                      "fields": [{{#text_fields}}"{{name}}^{{boost}}"{{/text_fields}}]
+                    }
+                  },
+                  "from": "{{from}}",
+                  "size": "{{size}}"
+                }
+                ""\",
+                "params": {
+                  "query_string": "My query string"
+                }
+              }
+            }""";
+
+        Map<String, Object> params = Map.of(
+            "text_fields",
+            List.of(Map.of("name", "title", "boost", 10), Map.of("name", "description", "boost", 2)),
+            "from",
+            0,
+            "size",
+            0
+        );
+
+        {
+            ParsingException e = expectThrows(ParsingException.class, () -> {
```

Comment:
This test appears to be passing, but for the wrong reason. The message from the exception thrown is:

```
org.elasticsearch.common.ParsingException: Unknown key for a START_OBJECT in [script].
```

Which reads like it's not even able to parse the script. Shouldn't the exception thrown here be `XContentParseException`?

Reply:
Oh, that's a good spot.  Interestingly, the test did fail without the fix, but it looks as though I'm not testing the specific scenario mentioned in the original issue.  I'll try and tidy up in a follow-up